### PR TITLE
RHCLOUD-31669 Pull in recently visited widget

### PR DESCRIPTION
**Pull request by aferd**

### CODE REVIEW

File: rest/models/DashboardTemplate.go [hunk 1]
```diff
@@ -48,15 +48,16 @@ const (
 	Ansible             AvailableWidgets = "ansible"
 	Rhel                AvailableWidgets = "rhel"
 	Openshift           AvailableWidgets = "openshift"
+	RecentlyVisited		AvailableWidgets = "recentlyVisited"
```
Comment (florkbr) [inline rest/models/DashboardTemplate.go:51]: @aferd can we fix the indentation of `AvailableWidgets` here?

File: rest/models/DashboardTemplate.go [hunk 2]
```diff
@@ -48,15 +48,16 @@ const (
...
 
 func (aw AvailableWidgets) IsValid() error {
 	switch aw {
-	case FavoriteServices, NotificationsEvents, LearningResources, ExploreCapabilities, Edge, Ansible, Rhel, Openshift:
+	case FavoriteServices, NotificationsEvents, LearningResources, ExploreCapabilities, Edge, Ansible, Rhel, Openshift, RecentlyVisited:
 		return nil
 	}
 
-	return fmt.Errorf("invalid widget. Expected one of [%s, %s, %s, %s, %s, %s, %s, %s] got %s", FavoriteServices, NotificationsEvents, LearningResources, ExploreCapabilities, Edge, Ansible, Rhel, Openshift, aw)
+	return fmt.Errorf("invalid widget. Expected one of [%s, %s, %s, %s, %s, %s, %s, %s] got %s", FavoriteServices, NotificationsEvents, LearningResources, ExploreCapabilities, Edge, Ansible, Rhel, Openshift, RecentlyVisited, aw)
```
Comment (florkbr) [inline rest/models/DashboardTemplate.go:60]: @aferd we need to add another `%s` placeholder to the string template for the new string you added:
```go
return fmt.Errorf("invalid widget. Expected one of [%s, %s, %s, %s, %s, %s, %s, %s, %s] got %s", FavoriteServices, NotificationsEvents, LearningResources, ExploreCapabilities, Edge, Ansible, Rhel, Openshift, RecentlyVisited, aw)
```

File: rest/models/DashboardTemplate.go [hunk 3]
```diff
@@ -199,15 +200,16 @@ type WidgetIcons string
...
 
 func (wi WidgetIcons) IsValid() error {
 	switch wi {
-	case BellIcon, StarIcon:
+	case BellIcon, StarIcon, HistoryIcon:
 		return nil
 	}
 
-	return fmt.Errorf("invalid widget icon. Expected one of %s, %s, got %s", BellIcon, StarIcon, wi)
+	return fmt.Errorf("invalid widget icon. Expected one of %s, %s, got %s", BellIcon, StarIcon, HistoryIcon, wi)
```
Comment (florkbr) [inline rest/models/DashboardTemplate.go:212]: @aferd same here, need another `%s`:  

```go
return fmt.Errorf("invalid widget icon. Expected one of %s, %s, %s got %s", BellIcon, StarIcon, HistoryIcon, wi)
```

File: rest/service/dashboardTemplate.go [hunk 4]
```diff
@@ -84,6 +84,14 @@ var (
 			Module:   "./OpenShiftWidget",
 			Defaults: models.BaseWidgetDimensions.InitDimensions(models.BaseWidgetDimensions{}, 2, 2, 4, 1),
 		},
+		models.RecentlyVisited: models.ModuleFederationMetadata{
```
Comment (florkbr) [inline rest/service/dashboardTemplate.go:87]: @aferd I've also been trying to keep our readme up to date with the various widgets. Do you mind adding `RecentlyVisited` here (to the `AvailableWidgets` example? https://github.com/RedHatInsights/chrome-service-backend/blob/main/docs/dashboard-layouts.md#widgets-mapping

File: rest/models/DashboardTemplate.go (hunk 1 above)
Comment (aferd) [reply rest/models/DashboardTemplate.go:51]: this is not showing on my side 😞 maybe if I fix it from here...

Comment (aferd) [inline rest/models/DashboardTemplate.go:51]:
```suggestion
	RecentlyVisited	        AvailableWidgets = "recentlyVisited"
```

File: rest/models/DashboardTemplate.go [hunk 5]
```diff
@@ -48,15 +48,16 @@ const (
 	Ansible             AvailableWidgets = "ansible"
 	Rhel                AvailableWidgets = "rhel"
 	Openshift           AvailableWidgets = "openshift"
+	RecentlyVisited	        AvailableWidgets = "recentlyVisited"
```
Comment (aferd) [inline rest/models/DashboardTemplate.go:51]:
```suggestion
	RecentlyVisited	    AvailableWidgets = "recentlyVisited"
```

Comment (aferd) [reply rest/models/DashboardTemplate.go:51]: fix spacing again

**Review by florkbr — APPROVED** [review]

**Pull request merged**